src/main/java/com/cisco/cta/taxii/adapter/AdapterConfiguration.java

```diff
@@ -99,7 +94,10 @@ public TaxiiPollResponseReaderFactory taxiiPollResponseReaderFactory() throws Ex
 
     @Bean
     public XMLInputFactory inputFactory() {
-        return XMLInputFactory.newFactory();
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
```

Comment:
I see you added protection against XXE. 👍 for that. Can you please also add a unit test to verify both the `XMLInputFactory` and `TransformerFactory` are actually resistant to such attack? Here is an example of the potentially malicious payload. If encountered, the deserialization process should fail.

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<!DOCTYPE Collection_Information_Request [
        <!ENTITY foo "bar">
        ]>
<taxii_11: ...
```